src/ImageSharp.Drawing/Processing/PathGradientBrush.cs

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.Primitives;
+using SixLabors.Shapes;
+
+namespace SixLabors.ImageSharp.Processing
+{
+    /// <summary>
+    /// Provides an implementation of a brush for painting gradients between multiple color positions in 2D coordinates.
+    /// It works similarly with the class in System.Drawing.Drawing2D of the same name.
+    /// </summary>
+    public sealed class PathGradientBrush : IBrush
+    {
+        private readonly Polygon path;
+
+        private readonly IList<Edge> edges;
+
+        private readonly Color centerColor;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PathGradientBrush"/> class.
+        /// </summary>
+        /// <param name="lines">Line segments of a polygon that represents the gradient area.</param>
+        /// <param name="colors">Array of colors that correspond to each point in the polygon.</param>
+        /// <param name="centerColor">Color at the center of the gradient area to which the other colors converge.</param>
+        public PathGradientBrush(ILineSegment[] lines, Color[] colors, Color centerColor)
+        {
+            this.path = new Polygon(lines);
+            this.centerColor = centerColor;
+
+            Color ColorAt(int index) => colors[index % colors.Length];
+
+            this.edges = this.path.LineSegments
+                .Select(s => new Path(s))
+                .Select((path, i) => new Edge(path, ColorAt(i), ColorAt(i + 1)))
+                .ToList();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PathGradientBrush"/> class.
+        /// </summary>
+        /// <param name="lines">Line segments of a polygon that represents the gradient area.</param>
+        /// <param name="colors">Array of colors that correspond to each point in the polygon.</param>
+        public PathGradientBrush(ILineSegment[] lines, Color[] colors)
+            : this(lines, colors, CalculateCenterColor(colors))
+        {
+        }
+
+        /// <inheritdoc />
+        public BrushApplicator<TPixel> CreateApplicator<TPixel>(
+            ImageFrame<TPixel> source,
+            RectangleF region,
+            GraphicsOptions options)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            return new PathGradientBrushApplicator<TPixel>(source, this.path, this.edges, this.centerColor, options);
+        }
+
+        private static Color CalculateCenterColor(Color[] colors) =>
+            new Color(colors.Select(c => c.ToVector4()).Aggregate((p1, p2) => p1 + p2) / colors.Length);
+
+        private static float DistanceBetween(PointF p1, PointF p2) => ((Vector2)(p2 - p1)).Length();
```

Comment:
Squared distance is enough for your purpose.  (`dx^2 + dy^2` instead of  `sqrt( dx^2 + dy^2 )`)

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.numerics.vector4.lengthsquared?view=netframework-4.8

Reply:
Upon inspection, `DistanceBetween` cannot be modified that way without affecting the output, since the brush relies on a ratio of actual distances.

I replaced `Length()` used in `FindIntersection` method as suggested, however, as it's only used to find the closest edge.